js/app.js

```diff
@@ -70,59 +75,69 @@ function scrollToItem(itemId) {
         blogSummary.querySelector(".blog-preview-toggle").dataset.item_id = blog.system.id;
-
-    const blogMediaImageURL = firstOrDefaultValue(blog, "blog_media___image", "url");
-	if (blogMediaImageURL)
-	{
-        const blogImage = document.importNode(blogImageTemplate.content, true);
-        blogImage.querySelector(".blog-image").src = blogMediaImageURL;
-        blogPost.insertBefore(blogImage, blogPost.childNodes[0]);
-    }
+    blogPost = insertBlogImage(blog, blogPost);
 
     const authorContent = firstOrDefaultContent(blog, modularContent, "author");
     const authorName = firstOrDefaultValue(authorContent, "full_name") || "(none)";
     blogSummary.querySelector(".summary-author").textContent = authorName;
-
-    const postDate = firstOrDefaultValue(blog, "post_date");
-    if(postDate) {
-        try {
-            const blogDate = new Date(postDate);
-            blogSummary.querySelector(".summary-date").textContent = blogDate.toDateString();
-        } catch(e) {
-            console.warn("Blog post date couldn't be parsed for " + JSON.stringify(blog));
-        }
-    }
+    blogSummary = insertBlogDate(blog, blogSummary);
 
     let body = firstOrDefaultValue(blog, "body") || "";
     body = convertImagesToHyperlink(body);
+    body = wrapTablesForOverflow(body);
 	blogSummary.querySelector(".summary-body").innerHTML = body;
 
     processContentSnippets(blogSummary, modularContent);
 	return blogSummary;
 }
-
-
+function insertBlogImage(blog, blogPost) {
+    const blogMediaImageURL = firstOrDefaultValue(blog, "blog_media___image", "url");
+    if (blogMediaImageURL) {
+        const blogImage = document.importNode(blogImageTemplate.content, true);
+        blogImage.querySelector(".blog-image").src = blogMediaImageURL;
+        blogPost.insertBefore(blogImage, blogPost.childNodes[0]);
+    }
+    return blogPost;
+}
+function insertBlogDate(blog, blogSummary) {
+    const postDate = firstOrDefaultValue(blog, "post_date");
+    if (postDate) {
+        const blogDate = new Date(postDate);
+        blogSummary.querySelector(".summary-date").textContent = blogDate.toDateString();
+    }
+    return blogSummary;
+}
 function convertImagesToHyperlink(blogBody) {
     const wrapper = document.createElement("div");
     wrapper.innerHTML = blogBody;
     wrapper.querySelectorAll("img").forEach(image => {
         const imageParentNode = image.parentNode;
-        const elementHyperlink = document.createElement('a');
+        const elementHyperlink = document.createElement("a");
         elementHyperlink.appendChild(image.cloneNode());
         elementHyperlink.href = image.src;
         elementHyperlink.target = "_blank";
         imageParentNode.replaceChild(elementHyperlink, image);
     });
     return wrapper.innerHTML;
 }
+function wrapTablesForOverflow(blogBody) {
+    const wrapper = document.createElement("div");
+    wrapper.innerHTML = blogBody;
+    wrapper.querySelectorAll("table").forEach(table => {
```

Comment:
the logic below will only work for tables which are immediate child elements of `wrapper` but this selector will find all of them (including e.g. tables in tables). Immediate child selector would be `> table`

Reply:
OK, will implement this before commit